### PR TITLE
feat(influxdb-scaler): add support for integer query results

### LINKED ISSUE #1973: [InfluxDB Scaler] Allow queries that return integers in addition to those that return floats
### Proposal

The InfluxDB scaler should continue to function if the query used returns an integer rather than a float. At the moment, there is an undocumented requirement that queries must return floats.

### Use-Case

I'd like to query values from InfluxDB that happen to be stored as integers (event counters and such) without having to remember to manually convert those values to floats in each Flux query.

### Anything else?

[influxdb_scaler.go lines 173 to 176](https://github.com/kedacore/keda/blob/03d0f98fb4b9c81e04be0dbc1069bb4a3a03f5b5/pkg/scalers/influxdb_scaler.go#L173-L176) contains the following code:

```golang
val, ok := result.Record().Value().(float64)
if !ok {
    return 0, fmt.Errorf("value could not be parsed into a float")
}
```

It seems like adding an additional branch that handles integers wouldn't be a huge burden.

### PR DESCRIPTION
**Pull request by philomory**
Signed-off-by: Adam Gardner <adam.gardner@magicmemories.com>

Adjusted the InfluxDB scaler so that it can successfully process queries that return integer results in addition to those that return float results.

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [X] Tests have been added
- [N/A] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [N/A] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated

Fixes #1973

### CODE REVIEW

**Comment by zroubalik** [comment]
@philomory could you please fix the linter problem? thx

**Comment by yquansah** [comment]
@philomory Yeah the linter problem the only hanging thing, other than that the PR looks good

File: pkg/scalers/influxdb_scaler.go [hunk 1]
```diff
@@ -170,9 +170,14 @@ func queryInfluxDB(queryAPI api.QueryAPI, query string) (float64, error) {
...
+	switch valRaw := result.Record().Value(); valRaw.(type) {
+	case float64:
+		val = valRaw.(float64)
+	case int64:
+		val = float64(valRaw.(int64))
+	default:
+		return 0, fmt.Errorf("value of type %T could not be converted into a float", valRaw)
 	}
 
 	return val, nil
```
Comment (ahmelsayed) [inline pkg/scalers/influxdb_scaler.go:183]: per the golangci-lint's error 

```
pkg/scalers/influxdb_scaler.go:174:2: typeSwitchVar: 2 cases can benefit from type switch with assignment (gocritic)

	switch valRaw := result.Record().Value(); valRaw.(type) {
```

```suggestion
	switch valRaw := result.Record().Value().(type) {
	case float64:
		return valRaw, nil
	case int64:
		return float64(valRaw), nil
	default:
		return 0, fmt.Errorf("value of type %T could not be converted into a float", valRaw)
```

**Comment by philomory** [comment]
Due to the unexpectedly early family arrival, I'm not near a workstation and probably won't be for some time. I could theoretically make changes to the PR from my iPad, but of course I can't run the linter _on_ the iPad and the workflow logs appear to have already been purged. If anyone wants to le me know what changes the linter is asking for I'd be happy to approve them (pre-emptively, even).

File: pkg/scalers/influxdb_scaler.go [hunk 2]
```diff
@@ -170,12 +170,13 @@ func queryInfluxDB(queryAPI api.QueryAPI, query string) (float64, error) {
...
-	}
-
-	return val, nil
+	switch valRaw := result.Record().Value().(type) {
+	case float64:
+		return valRaw, nil
+	case int64:
+		return float64(valRaw), nil
+	default:
+		return 0, fmt.Errorf("value of type %T could not be converted into a float", valRaw)
```
Comment (ahmelsayed) [inline pkg/scalers/influxdb_scaler.go:179]:
```suggestion
		return 0, fmt.Errorf("value of type %T could not be converted into a float", valRaw)
		}
```

**Comment by ahmelsayed** [comment]
@philomory sorry I missed a curly bracket :). 
And no worries, enjoy family time!

**Comment by zroubalik** [comment]
The DCO signoff can be fixed when merging this PR, it is not a problem.

File: pkg/scalers/influxdb_scaler.go [hunk 3]
```diff
@@ -170,12 +170,14 @@ func queryInfluxDB(queryAPI api.QueryAPI, query string) (float64, error) {
...
+		}
```
Comment (zroubalik) [inline pkg/scalers/influxdb_scaler.go:180]:
```suggestion
	}
```
Linter is complaining about indentation :(

**Review by zroubalik — APPROVED** [review]

**Pull request merged**